include/deal.II/base/symmetric_tensor.h

```diff
@@ -1215,17 +1215,15 @@ namespace internal
       {
       case 1:
         return data[0] * sdata[0];
-      case 2:
-        return (data[0] * sdata[0] +
-                data[1] * sdata[1] +
-                NumberType<Number>::value(2.0) * data[2] * sdata[2]);
```

Comment:
Are you removing this block because the compiler has an easier time vectorizing the loop below than the straight-line code here?